src/ngraph/runtime/cpu/builder/sigmoid.cpp

```diff
@@ -109,8 +114,132 @@ namespace ngraph
                 functors.emplace_back(functor);
             }
 
+            template <typename ElementType>
+            static Eigen::TensorMap<Eigen::Tensor<ElementType, 1, Eigen::RowMajor>>
+                wrap_into_tensor_map(void* data, size_t tensor_size)
+            {
+                Eigen::array<Eigen::Index, 1> dims;
+                dims[0] = tensor_size;
+
+                Eigen::TensorMap<Eigen::Tensor<ElementType, 1, Eigen::RowMajor>> out(
+                    static_cast<ElementType*>(data), dims);
+                return out;
+            }
+
+            template <>
+            void Builder::BUILDER_DECL(ngraph::op::SigmoidMultiply)
+            {
+                auto& functors = external_function->get_functors();
+                auto& tensor_data = external_function->get_tensor_data();
+
+                auto& arg0_tensor = tensor_data[args[0].get_name()];
+                auto& arg1_tensor = tensor_data[args[1].get_name()];
+                auto& out_tensor = tensor_data[out[0].get_name()];
+                auto tensor_size = shape_size(args[0].get_shape());
+
+                auto sigmoid_mul = static_cast<const ngraph::op::SigmoidMultiply*>(node);
+                function<void(CPURuntimeContext*)> functor;
+
+                const size_t index =
+                    static_cast<size_t>(sigmoid_mul->get_input_func_type(0)) *
+                        static_cast<size_t>(ngraph::op::SigmoidMultiply::FunctionType::NumTypes) +
+                    static_cast<size_t>(sigmoid_mul->get_input_func_type(1));
+
+                switch (index)
+                {
+                case 0 /*Logistic|Logistic*/:
+                    functor = [&, tensor_size](CPURuntimeContext* ctx) {
+                        auto in0 = wrap_into_tensor_map<float>(arg0_tensor, tensor_size);
+                        auto in1 = wrap_into_tensor_map<float>(arg1_tensor, tensor_size);
+                        auto out_tm = wrap_into_tensor_map<float>(out_tensor, tensor_size);
+                        auto c = (in0.exp() * in1.exp()) / ((in0.exp() + 1.f) * (in1.exp() + 1.f));
```

Comment:
Some of these expressions could probably be optimized but this looks like something we'd JIT anyway I guess...

Reply:
I'd just leave it as is. SigmoidMultiply doesn't seem to be the most important fusion, so this is probably good enough for now. I don't want to spend a lot of time on the code that we are going to scratch in the near future as soon as we get a compiler.

Reply:
Sounds good